src/rust/src/x509/certificate.rs

```diff
@@ -193,9 +193,17 @@ impl Certificate {
         let val = self.raw.borrow_value();
         let mut tbs_precert = val.tbs_cert.clone();
         // Remove the SCT list extension
-        match tbs_precert.extensions {
-            Some(extensions) => {
-                let readable_extensions = extensions.unwrap_read().clone();
+        match val.tbs_cert.extensions() {
+            Err(oid) => {
+                let oid_obj = oid_to_py_oid(py, &oid)?;
+                Err(exceptions::DuplicateExtension::new_err((
+                    format!("Duplicate {} extension found", oid),
+                    oid_obj.into_py(py),
+                ))
+                .into())
+            }
+            Ok(Some(extensions)) => {
+                let readable_extensions = extensions.as_ref().unwrap_read().clone();
```

Comment:
If this is the use case for `as_ref()`, I wouldn't use the trait, just add a regular method

Reply:
Done; made it `as_raw`.